### PR TITLE
[risk=low][RW-9007] Crude UI for Create/Delete Cromwell

### PR DESCRIPTION
**Pull request by jmthibault79**
Crude early preview to unblock other work.  Polishing will come later.  This is feature-flagged so the UI will not be going to Prod.

Launch Button (Create) and Trash Can (Delete) work, but they do not yet provide indication of progress.

Closing and reopening the Apps Panel will show the app's status string, as Leonardo sees it.

Initial opening of Apps Panel
Not Running

After expanding Cromwell
Not Running Expanded

After clicking Launch and closing/reopening the Apps Panel
Provisioning

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/components/apps-panel.tsx [hunk 1]
```diff
@@ -92,59 +99,60 @@ export const AppsPanel = (props: {
     !appStates.find((s) => s.appType === appType)?.shouldExpandByDefault;
   const showAvailableSection = appsToDisplay.some(showInAvailableSection);
 
-  const ActiveApps = (sectionProps: { showCloseButtonHere: boolean }) => (
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel.tsx:95]: inlined.  See #7304 for that change alone

File: ui/src/app/components/help-sidebar-icons.tsx [hunk 2]
```diff
@@ -122,18 +122,11 @@ const displayRuntimeStatusIcon = (
     (aa) => aa.appType === UIAppType.JUPYTER
   );
 
-  const containerStyle: CSSProperties = showAppsPanel
-    ? {
-        height: '100%',
-        alignItems: 'center',
-        justifyContent: 'space-around',
-        background: colors.white,
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar-icons.tsx:130]: removed this distinction - looks better

File: ui/src/app/services/swagger-fetch-clients.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@
  */
 
 import {
-  AuthDomainApi,
```
Comment (jmthibault79) [inline ui/src/app/services/swagger-fetch-clients.ts:24]: not used

File: ui/src/app/services/swagger-fetch-clients.ts [hunk 4]
```diff
@@ -109,6 +108,7 @@ export const cohortsApi = bindCtor(CohortsApi);
 export const conceptSetsApi = bindCtor(ConceptSetsApi);
 export const configApi = bindCtor(ConfigApi);
 export const dataSetApi = bindCtor(DataSetApi);
+export const disksApi = bindCtor(DisksApi);
```
Comment (jmthibault79) [inline ui/src/app/services/swagger-fetch-clients.ts:111]: alphabetized

File: ui/src/app/components/apps-panel/utils.tsx [hunk 5]
```diff
@@ -25,4 +38,55 @@ export const appAssets: AppAssets[] = [
...
     icon: rStudioIcon,
   },
+  {
+    appType: UIAppType.CROMWELL,
+    logo: cromwellLogo,
+    icon: cromwellIcon,
+  },
 ];
+
+// TODO replace with better defaults
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/utils.tsx:48]: These work as a POC for create/delete so I don't think this is blocking.  But we should update this ASAP.

File: ui/src/app/components/apps-panel/utils.tsx [hunk 6]
```diff
@@ -25,4 +38,55 @@ export const appAssets: AppAssets[] = [
...
+  },
+  persistentDiskRequest: {
+    size: 10,
+    diskType: DiskType.Standard,
+  },
+};
+
+// visible/actionable logic is copied from runtime-utils
+
+export const isVisible = (status: AppStatus): boolean =>
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/utils.tsx:64]: "visible" meaning "do we even tell the user that such a thing exists"

File: ui/src/app/components/help-sidebar-icons.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/components/help-sidebar-icons.tsx:130]: This change goes with the new Jupyter-icon below

File: ui/src/testing/stubs/apps-api-stub.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { AppsApi, EmptyResponse, ListAppsResponse } from 'generated/fetch';
+
+import { stubNotImplementedError } from 'testing/stubs/stub-utils';
+
+export class AppsApiStub extends AppsApi {
```
Comment (jmthibault79) [inline ui/src/testing/stubs/apps-api-stub.ts:5]: Strictly boilerplate: implement only those stubs which the UI actually calls

File: ui/src/app/components/apps-panel.tsx [hunk 8]
```diff
@@ -52,23 +53,28 @@ export const AppsPanel = (props: {
...
+  // all apps besides Jupyter
+  const [userApps, setUserApps] = useState<UserAppEnvironment[]>();
+  useEffect(() => {
+    appsApi().listAppsInWorkspace(workspace.namespace).then(setUserApps);
+  }, []);
 
   const appStates = [
     {
       appType: UIAppType.JUPYTER,
-      expandable: true,
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel.tsx:64]: no longer a need to have "unexpandable" app-types

File: api/src/main/resources/workbench-api.yaml [hunk 9]
```diff
@@ -6393,15 +6393,15 @@ definitions:
...
+    type: object
+    required:
+      - appType
+      - kubernetesRuntimeConfig
+      - persistentDiskRequest
     description: Create User App request body
     properties:
       appType:
         "$ref": "#/definitions/AppType"
-      autopauseThreshold:
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/src/main/resources/workbench-api.yaml:6420]: We would need this eventually. https://broadworkbench.atlassian.net/browse/IA-3706 is tracking the support of autopause. 
Do you plan to remove this for now, and added it back later once we worked on this feature. Or we should just keep as not used?

File: ui/src/app/components/apps-panel/utils.tsx [hunk 10]
```diff
@@ -25,4 +38,55 @@ export const appAssets: AppAssets[] = [
...
+// TODO replace with better defaults
+export const defaultCromwellConfig: CreateAppRequest = {
+  appType: AppType.CROMWELL,
+  kubernetesRuntimeConfig: {
+    numNodes: 1,
+    machineType: DEFAULT_MACHINE_NAME,
+    autoscalingEnabled: false,
+  },
+  persistentDiskRequest: {
+    size: 10,
```
Comment (yonghaoy) [inline ui/src/app/components/apps-panel/utils.tsx:57]: Is 10 too small for this app? I usually use 500GB, but I never test 10 before.

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 11]
```diff
@@ -41,53 +54,163 @@ const styles = reactStyles({
...
   onClickDeleteRuntime: Function;
-}) => {
+}
+export const ExpandedApp = (props: ExpandedAppProps) => {
   const { runtime } = useStore(runtimeStore);
-  const { appType, workspace, onClickRuntimeConf, onClickDeleteRuntime } =
-    props;
+  const {
+    appType,
+    initialUserAppInfo,
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/expanded-app.tsx:154]: TODO: keep the user app info updated.  Here we only check at AppsPanel render time.

**Review by yonghaoy — APPROVED** [review]

File: api/src/main/resources/workbench-api.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/resources/workbench-api.yaml:6420]: We should add this back in after Leo supports it.

But today it's not true to say that we support `autopauseThreshold` for User Apps, so the API should reflect that.

File: ui/src/app/components/apps-panel/utils.tsx (hunk 10 above)
Comment (jmthibault79) [reply ui/src/app/components/apps-panel/utils.tsx:57]: I have no idea what values are correct here :)

If you have better defaults, I'm happy to use them, in this PR or later.

File: ui/src/app/components/apps-panel/utils.tsx (hunk 5 above)
Comment (jmthibault79) [reply ui/src/app/components/apps-panel/utils.tsx:48]: Marcus will ask around for better values for this

File: ui/src/app/components/apps-panel/utils.tsx (hunk 10 above)
Comment (jmthibault79) [reply ui/src/app/components/apps-panel/utils.tsx:57]: Update: Marcus is checking on what disk size is appropriate

**Comment by jmthibault79** [comment]
We talked online.  Looks like the issue was that it took a little bit of time (not instant) to transition but the changes eventually appear.  That's good enough for this iteration.

**Pull request merged**